server/src/main/java/org/elasticsearch/action/search/SearchScrollRequest.java

```diff
@@ -83,6 +83,10 @@ public SearchScrollRequest scrollId(String scrollId) {
         return this;
     }
 
+    public ParsedScrollId parsedScrollId() {
+        return TransportSearchHelper.parseScrollId(scrollId);
```

Comment:
In principle, I would shy away from getters that do any processing, unless the method has a verb in the name.

Reply:
Given `TransportSearchHelper.parseScrollId(scrollId)` can throw exception, I think it is fair to name it with a verb. I updated to be `parseScrollId` (same name as `TransportSearchHelper#parseScrollId`). The extra layer of proxy (from ParsedScrollId to TransportSearchHelper) is because `TransportSearchHelper` is not public and it feels better to keep it that way.